1-utils.c

```diff
@@ -0,0 +1,56 @@
+#include "shell.h"
+
+/**
+ * _strncmp - Function that compare two strings
+ * @str1: first string
+ * @str2: second string
+ * @n: amount of bytes to compare
+ *
+ * Return: 0 if second string is found in first string.
+ */
+int _strncmp(const char *str1, const char *str2, size_t n)
+{
+	while (n > 0)
+	{
+		if (*str1 != *str2)
+			return (*str1 - *str2);
+		if ((*str1 == '\0') || (*str2 == '\0'))
+			break;
+		str1++;
+		str2++;
+		n--;
+	}
+
+	return 0;
+}
+
+/**
+ * _strcat_delim - concatenate two strings with the delimiter "/"
+ * @dest: destination of the concatenated string
+ * @str1: first string
+ * @str2: second string
+ *
+ * Return: retrun concatenated string
+ */
+char *_strcat_delim(char *dest, char *str1, char *str2)
+{
+	size_t str1_len = _strlen(str1);
+	char delim = '/';
+	size_t dest_len = 0;
+
+	_strcpy(dest, str1);
+
+	/*check if the delimiter can be added*/
+	if (str1_len > 0 && delim != '\0')
```

Comment:
Why check if `delim != \0` if it has already been declared on line 38 and would not be passed as an argument? It will be better to remove it and instead make `char delim` -> `const char delim`

Reply:
Unless you want it passed as a variable then in that case, this particular check would be valid

Reply:
i will make it const then